.github/workflows/release.yml

```diff
@@ -14,36 +23,72 @@ jobs:
           java-version: 17
           distribution: 'temurin'
           cache: 'gradle'
-      - name: Grant execute permission for gradlew
-        run: chmod +x gradlew
+      # Setup Gradle
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
       - name: Build with Gradle
         run: ./gradlew build
-      - name: Get the version
-        id: get_version
-        run: echo ::set-output name=VERSION::${GITHUB_REF#refs/tags/}
+      - name: Tag Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config user.email "action@github.com"
+          git config user.name "GitHub Action"
+          if git diff --quiet; then
+            echo "No changes to commit."
+          else
+            git commit -sam "chore(skip-release): set version to ${{ inputs.release_version }}"
+          fi
+          git tag ${{ inputs.release_version }}
+          git push origin ${{ inputs.release_version }}
+      - name: Set Release Version
+        shell: bash
+        run: |
+          CURRENT_VERSION=$(grep "projectVersion=" gradle.properties | cut -d'=' -f2)
+          NEW_VERSION=${{ inputs.release_version }}
+          awk -v current="$CURRENT_VERSION" -v new="$NEW_VERSION" 'BEGIN {FS=OFS="="} $1 == "projectVersion" { $2 = new }1' gradle.properties > tmpfile && mv tmpfile gradle.properties
+          echo "Release version: $NEW_VERSION"
+          echo "PLUGIN_VERSION=${NEW_VERSION}" >> $GITHUB_ENV
+      # Publish the plugin to local release repo
+      - name: Publish Library
+        run: |
+          ./gradlew publish
+          echo "Published $PLUGIN_VERSION."
+      # Set next SNAPSHOT version
+      - name: Increment Plugin Version
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          CURRENT_VERSION=$(grep "projectVersion=" gradle.properties | cut -d'=' -f2)
+          IFS="-" read -ra VERSION_PARTS <<< "$CURRENT_VERSION"
+          IFS="." read -ra VERSION_NUM <<< "${VERSION_PARTS[0]}"
+          ((VERSION_NUM[2]+=1))
+          NEW_VERSION="${VERSION_NUM[0]}.${VERSION_NUM[1]}.${VERSION_NUM[2]}-SNAPSHOT"
```

Comment:
@sbouchet dont fully get the logic here: if I understand it correctly, we now manually release using a version that we manually provide. 

```bash
NEW_VERSION=${{ inputs.release_version }}
```
Shouldn't we use this manually provided version to bump and set as new version? The script seems to use the current project version instead? What am I missing?

```
CURRENT_VERSION=$(grep "projectVersion=" gradle.properties | cut -d'=' -f2)
...
NEW_VERSION="${VERSION_NUM[0]}.${VERSION_NUM[1]}.${VERSION_NUM[2]}-SNAPSHOT"
```

Reply:
on the steps above, we modified the gradle.properties file to use the input version ( before building it ) .so this step is to increment micro and set to SNAPSHOT.
that's equivalent to use directly the manually provided version